### PR TITLE
 Fixes #230 set value origin to unknow after change 

### LINKED ISSUE #230: Set value origin mode Unknown at manual changes
see https://github.com/Open-Systems-Pharmacology/OSPSuite.Core/issues/231
This is the MoBi implementation

### PR DESCRIPTION
**Pull request by msevestre**
Big PR unfortunately but there was a lot of work that needed to be done in MoBi to be able to set the pace

### CODE REVIEW

File: src/MoBi.Presentation/Tasks/MoBiFormulaTask.cs [hunk 1]
```diff
@@ -129,32 +163,55 @@ public IMoBiCommand ChangeVariableName(SumFormula formula, string newVariableNam
...
-         return new ChangeTableFormulaWithOffsetTableObjectPathCommand(formula, path, buildingBlock).Run(_context);
+         var command = new ChangeTableFormulaWithOffsetTableObjectPathCommand(formula, path, buildingBlock).Run(_context);
+         return withUpdatedDefaultStateAndValueOrigin(command, formula, buildingBlock);
       }
 
       public IMoBiCommand SetConstantFormulaValue(ConstantFormula formula, double kernelValue, Unit newDisplayUnit, Unit oldDisplayUnit, IBuildingBlock buildingBlock, IEntity formulaOwner)
       {
-         return new SetConstantFormulaValueCommand(formula, kernelValue, newDisplayUnit, oldDisplayUnit, buildingBlock, formulaOwner).Run(_context);
+         var command = new SetConstantFormulaValueCommand(formula, kernelValue, newDisplayUnit, oldDisplayUnit, buildingBlock, formulaOwner).Run(_context);
+         return withUpdatedDefaultStateAndValueOrigin(command, formula, buildingBlock);
```
Comment (msevestre) [inline src/MoBi.Presentation/Tasks/MoBiFormulaTask.cs]: I think those are all the methods where update of state and value origin for depending parmaeters are required.... to check

**Comment by msevestre** [comment]
@Yuri05 I am going to need that merged soon so that I can get going. Can you review today? Or do you want to review later?

**Comment by msevestre** [comment]
I am starting on https://github.com/Open-Systems-Pharmacology/OSPSuite.Core/issues/246 now so this can wait for a few more days

File: src/MoBi.Presentation/Tasks/MoBiFormulaTask.cs [hunk 2]
```diff
@@ -175,28 +232,29 @@ public IFormula CreateNewFormula(Type formulaType, IDimension formulaDimension)
...
 
-         if (formulaType == typeof (TableFormula))
+         if (formulaType == typeof(TableFormula))
             return createTableFormula();
 
-         if (formulaType == typeof (SumFormula))
+         if (formulaType == typeof(SumFormula))
             return _context.Create<SumFormula>();
 
          if (formulaType == typeof(TableFormulaWithOffset))
```
Comment (Yuri05) [inline src/MoBi.Presentation/Tasks/MoBiFormulaTask.cs]: case `DistributionFormula `not required here?

Comment (msevestre) [reply src/MoBi.Presentation/Tasks/MoBiFormulaTask.cs]: That's the old code here . Nothing to do with the PR

**Pull request merged**